core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -457,9 +457,9 @@ private[spark] object HadoopRDD extends Logging {
       if (locationStr != "localhost") {
         if (loc.isInMemory) {
           logDebug(s"Partition $locationStr is cached by Hadoop.")
-          Some(HDFSCacheTaskLocation(locationStr).toString)
+          Option(HDFSCacheTaskLocation(locationStr).toString)
         } else {
-          Some(HostTaskLocation(locationStr).toString)
+          Option(HostTaskLocation(locationStr).toString)
```

Comment:
and this seems can't be `null` too because of `val hstr = str.stripPrefix(inMemoryLocationTag)` in the `apply`.